pkg/git/git.go

```diff
@@ -281,3 +286,35 @@ func ValidateGitSSHURLFormat(url string) bool {
 	}
 	return false
 }
+
+func configSparseCheckout(logger *zap.SugaredLogger, spec FetchSpec) error {
+	if spec.SparseCheckoutDirectories != "" {
+		if _, err := run(logger, "", "config", "core.sparsecheckout", "true"); err != nil {
+			return err
+		}
+
+		dirPatterns := strings.Split(spec.SparseCheckoutDirectories, spec.SparseCheckoutDelimiter)
+
+		cwd, err := os.Getwd()
+		if err != nil {
+			return err
```

Comment:
Any reason to use `logger.Errorf` for the other errors in this func except for this one?  Would it be worth logging this in a similar way?

Reply:
I think it would be worth adding to maintain consistency.